snapd-qt/Snapd/client.h

```diff
@@ -899,6 +899,51 @@ class Q_DECL_EXPORT QSnapdDownloadRequest : public QSnapdRequest
     Q_DECLARE_PRIVATE(QSnapdDownloadRequest)
 };
 
+class QSnapdCheckThemesRequestPrivate;
+class Q_DECL_EXPORT QSnapdCheckThemesRequest : public QSnapdRequest
+{
+    Q_OBJECT
+
+public:
+    enum ThemeStatus
+    {
+       ThemeInstalled,
+       ThemeAvailable,
+       ThemeUnavailable
+    };
+    Q_ENUM(ThemeStatus)
+
+    explicit QSnapdCheckThemesRequest (const QStringList& gtkThemeNames, const QStringList& iconThemeNames, const QStringList& soundThemeNames, void *snapd_client, QObject *parent = 0);
+    ~QSnapdCheckThemesRequest ();
+    virtual void runSync ();
+    virtual void runAsync ();
+    Q_INVOKABLE ThemeStatus gtkThemeStatus (const QString& name) const;
+    Q_INVOKABLE ThemeStatus iconThemeStatus (const QString& name) const;
+    Q_INVOKABLE ThemeStatus soundThemeStatus (const QString& name) const;
+    void handleResult (void *, void *);
```

Comment:
You could avoid exposing this method, if you kept a pointer to `QSnapdCheckThemesRequest` in `QSnapdCheckThemesRequestPrivate`.

Reply:
Ah good point. All the existing classes follow this pattern, so I guess the damage is already done :(